### PR TITLE
Added code for week 6 check in.

### LINKED ISSUE #13: Major: User Interface
User interface needs to be created so that the user can actually interact with a mobile app. At this time, Kivy seems like the best bet for that and that's my plan. 

Sketches for that can be found in my deliverables.

### LINKED ISSUE #25: Addition of URL dataframe columns
Want to append the URL and the matches for both debugging purposes and for user's ease of access. Putting this here so I don't forget.

### LINKED ISSUE #28: D5: User testing: Add options that aren't vegan
It's been fun testing with just vegan recipes, but it's time to add more recipes for the users sake. Many users expressed wanting to use meat... Me too.

### LINKED ISSUE #30: D5: End to End Testing: Division by 0
ZeroDivisionError: division by zero 
Trivial bug (I always say this and then eat my words) 

Line 145: fraction = (matches.count(True) / len(ingredientList)) * 100 needs a catch for if it's dividing by 0

### LINKED ISSUE #31: D5: End-to-end: NoneType
While scraping, encountered the following error: 
AttributeError: 'NoneType' object has no attribute 'replace'

### PR DESCRIPTION
**Pull request by schweinsbergs**
Just code for week 6 check in. I need to clean this up again... 

Added: 
A user input window 
Ability to visual dataframe

### CODE REVIEW

**Comment by schweinsbergs** [comment]
Fixes #13

**Comment by schweinsbergs** [comment]
Attempted to add non-vegan recipes to the dataframe, but the website that I used appears to have a LOT of empty information columns. Not good. Gotta keep pushing on. 

This recent push fixes #30 -- Added a case for if there are 0 matches so the system doesn't freak out dividing by 0.

**Comment by schweinsbergs** [comment]
Last push fixed: 

#31 
#28 
#25

File: Code/recipeApp.py [hunk 1]
```diff
@@ -105,35 +107,29 @@ def open_pickle_jar(self):
...
-        self.user_ingredients = userinput.split()  # splits each ingredient up
-        # print("ingredients:", self.user_ingredients)
 
     def compare_ingredients(self):
         """
         Compares the dataframe's ingredients list and the user's.
         This might be helpful: https://towardsdatascience.com/dealing-with-list-values-in-pandas-dataframes-a177e534f173
 
-        https://www.datacamp.com/community/tutorials/fuzzy-string-python
+        https://stackoverflow.com/questions/35240528/reverse-dataframes-rows-order-with-pandas
```
Comment (alfarozavalae) [inline Code/recipeApp.py:116]: Are these sources? Make sure to say so if they are.

File: Code/recipeApp.py [hunk 2]
```diff
@@ -142,49 +138,39 @@ def compare_ingredients(self):
...
+               # print(specficmatches)
+
 
-           fraction = (matches.count(True) / len(ingredientList)) * 100
+
+           # Calculate the percentage matched here by calculating the length, put that on the dataframe, and reverse it.
+           if matches.count(True) == 0:
+               fraction = 0
+           else:
+            fraction = (matches.count(True) / len(ingredientList)) * 100
```
Comment (alfarozavalae) [inline Code/recipeApp.py:151]: This seems to work to accomplish what you stated in issue #30 
Good job!

File: venv/Lib/site-packages/pip/_internal/cli/parser.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        section_items = {
+            name: [] for name in override_order
+        }  # type: Dict[str, List[Tuple[str, Any]]]
+        for section_key, val in self.config.items():
+            # ignore empty values
+            if not val:
+                logger.debug(
+                    "Ignoring configuration key '%s' as it's value is empty.",
+                    section_key,
+                )
```
Comment (alfarozavalae) [inline venv/Lib/site-packages/pip/_internal/cli/parser.py:202]: Assuming this is the code for issue #31 
Good job fixing this bug!

**Review by alfarozavalae — APPROVED** [review]
I have added comments to your issues and along the document too. 
This PR is approved so you can merge this code now.

**Pull request merged**